rep-2001.rst

```diff
@@ -38,26 +38,45 @@ We define three main entry points for ROS users.
  * desktop (recommended)
  * ros_base
  * ros_core
+ * ros_minimal
 
 Variants
 --------
 
+ROS Minimal
+'''''''''''
+
+The `ros_minimal` variant is a highly restricted set of ROS2 functionality.
+It targets resource- and tool-constrained environments where package managers may be limited or unavailable.
+The intention is that this variant can target cross-compiling for custom OS sysroots that may not have build tools or Python available that run on the target platform.
+It may not contain any GUI or Python dependencies.
+It may not have any dependencies outside a source workspace of its members with an exception of ubiquitous Linux libraries.
+Note that this variant explicitly does not include an implementation of either RMW or rcl_logging, users will need to choose their own that handle the dependencies that come with them.
```

Comment:
I think we need to at least have one RMW in order to be able to build this variant - it can be just the default (FastRTPS) - or if that requires undesired dependencies any other RMW impl.

Reply:
The attempt to avoid FastRTPS was because of its dependency on libasio-dev. I'm more inclined toward Cyclone because it has no external rosdeps, but I know it's not the default RMW - what do you think?

Reply:
I included `rcl_spdlog` and `rmw_cyclonedds` in the latest revision

Reply:
I think there needs to be a paragraph describing the rational why CycloneDDS was chosen over the default RMW FastRTPS.

Reply:
FWIW, and I'm not advocating for FastRTPS, I've been able to cross compile to an embedded ARM target using FastRTPS via `--cmake-flags -DTHIRDPARTY=ON` which forces FastRTPS to source it's own copy of asio (among other things).

Reply:
That's good information - perhaps we can target this at "one of" like the current core does